src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -0,0 +1,278 @@
+            if (zoneRules.isFixedOffset()) {
+              data.add(
+                  new HostColumnVector.StructData(Long.MIN_VALUE, Long.MIN_VALUE,
+                      zoneRules.getOffset(Instant.now()).getTotalSeconds())
+              );
+            } else {
+              // Capture the first official offset (before any transition) using Long min
+              ZoneOffsetTransition first = transitions.get(0);
+              data.add(
+                  new HostColumnVector.StructData(Long.MIN_VALUE, Long.MIN_VALUE,
+                      first.getOffsetBefore().getTotalSeconds())
+              );
+              transitions.forEach(t -> {
+                // Whether transition is an overlap vs gap.
+                // In Spark:
+                // if it's a gap, then we use the offset after *on* the instant
+                // If it's an overlap, then there are 2 sets of valid timestamps in that are overlapping
+                // So, for the transition to UTC, you need to compare to instant + {offset before} 
+                // The time math still uses {offset after}
+                if (t.isGap()) {
+                  data.add(
+                      new HostColumnVector.StructData(
+                          t.getInstant().getEpochSecond(),
+                          t.getInstant().getEpochSecond() + t.getOffsetAfter().getTotalSeconds(),
+                          t.getOffsetAfter().getTotalSeconds())
+                  );
+                } else {
+                  data.add(
+                      new HostColumnVector.StructData(
+                          t.getInstant().getEpochSecond(),
+                          t.getInstant().getEpochSecond() + t.getOffsetBefore().getTotalSeconds(),
+                          t.getOffsetAfter().getTotalSeconds())
+                  );
+                }
+              });
+            }
+            masterTransitions.add(data);
+            zoneIdToTable.put(zoneId.getId(), idx);
+          }
+        }
+        HostColumnVector.DataType childType = new HostColumnVector.StructType(false,
+            new HostColumnVector.BasicType(false, DType.INT64),
+            new HostColumnVector.BasicType(false, DType.INT64),
+            new HostColumnVector.BasicType(false, DType.INT32));
+        HostColumnVector.DataType resultType =
+            new HostColumnVector.ListType(false, childType);
+        HostColumnVector fixedTransitions = HostColumnVector.fromLists(resultType,
+            masterTransitions.toArray(new List[0]));
+        fixedTransitionsFuture.complete(fixedTransitions);
+        zoneIdToTableFuture.complete(zoneIdToTable);
+      } catch (Exception e) {
+        fixedTransitionsFuture.completeExceptionally(e);
+        zoneIdToTableFuture.completeExceptionally(e);
+        throw e;
+      }
+    }
+  }
+
+  private void close() {
+    try (HostColumnVector hcv = getHostFixedTransitions()) {
+      // automatically closed
+    }
+  }
```

Comment:
is not this try-with-resources equivalent to 
```suggestion
  private void close() {
      getHostFixedTransitions().close()
  }
```

either  
- getHostFixedTransitions() throws and then there is nothing to call close on
- or it does not then it will be called in the implicit "finally"

neither the exception from getHostFixedTransitions nor from getHostFixedTransitions().close() are caught